packages/toolkit/src/view/knowledge/components/KnowledgeBaseTab.tsx

```diff
@@ -0,0 +1,135 @@
+import { Separator, Skeleton } from "@instill-ai/design-system";
+import * as React from "react";
+import { KnowledgeBaseCard } from "./KnowledgeBaseCard";
+import { CreateKnowledgeBaseCard } from "./CreateKnowledgeBaseCard";
+import { CreateKnowledgeDialog } from "./CreateKnowledgeDialog";
+import {
+  useGetKnowledgeBases,
+  useCreateKnowledgeBase,
+} from "../../../lib/react-query-service/knowledge";
+import {
+  InstillStore,
+  useAuthenticatedUser,
+  useInstillStore,
+  useShallow,
+} from "../../../lib";
+import { KnowledgeBase } from "../../../lib/vdp-sdk/knowledge/types";
+
+type KnowledgeBaseTabProps = {
+  onKnowledgeBaseSelect: (knowledgeBase: KnowledgeBase) => void;
+  onDeleteKnowledgeBase: (knowledgeBase: KnowledgeBase) => void;
+  accessToken: string | null;
+};
+
+export const KnowledgeBaseTab = ({
+  onKnowledgeBaseSelect,
+  //   onDeleteKnowledgeBase,
+  accessToken,
+}: KnowledgeBaseTabProps) => {
+  const [loading, setLoading] = React.useState(false);
+
+  const selector = (store: InstillStore) => ({
+    accessToken: store.accessToken,
+    enabledQuery: store.enabledQuery,
+  });
+
+  const { enabledQuery } = useInstillStore(useShallow(selector));
+
+  const me = useAuthenticatedUser({
+    enabled: enabledQuery,
+    accessToken,
+  });
+  const userID = me.isSuccess ? me.data.id : null;
+
+  const [isCreateDialogOpen, setIsCreateDialogOpen] = React.useState(false);
+  const [knowledgeBases, setKnowledgeBases] = React.useState<KnowledgeBase[]>(
+    []
+  );
+  const getKnowledgeBases = useGetKnowledgeBases({
+    accessToken,
+    uid: userID as string,
```

Comment:
You can and probably should use `<Nullable<string>>` in the `useGetKnowledgeBases` to avoid type casting.